### PR TITLE
chore(typings): upgrade 'tsd' to 'typings'

### PR DESCRIPTION
**Pull request by NathanWalker**
This is mostly done, however there are several type warnings still... it does build and works though.

@gdi2290 I would greatly appreciate your help in finalizing this if you could checkout this typings branch and clear the remaining type warnings (stuck on those):
https://github.com/NathanWalker/angular2-seed/tree/typings

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,13 +11,14 @@
     "build.prod": "gulp build.prod",
     "build.test": "gulp build.test",
     "build.test.watch": "gulp build.test.watch",
+    "clean-ng2": "rimraf node_modules/angular2/manual_typings",
```
Comment (gdi2290) [inline package.json]: you also need to include `node_modules/angular2/typings` to be removed

File: typings.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "ambientDependencies": {
```
Comment (gdi2290) [inline typings.json]: you're also missing the `ng2` type info 
https://github.com/gdi2290/typings-ng2
as
`"ng2": "github:gdi2290/typings-ng2/ng2.d.ts#32998ff5584c0eab0cd9dc7704abb1c5c450701c",`

**Comment by NathanWalker** [comment]
Thanks @gdi2290 I updated per your suggestions, however I still get these warnings:
```
app/about/components/about.e2e.ts(1,1): error TS2304: Cannot find name 'describe'.
app/about/components/about.e2e.ts(3,3): error TS2304: Cannot find name 'beforeEach'.
app/about/components/about.e2e.ts(4,5): error TS2304: Cannot find name 'browser'.
app/about/components/about.e2e.ts(7,3): error TS2304: Cannot find name 'it'.
app/about/components/about.e2e.ts(8,5): error TS2304: Cannot find name 'expect'.
app/about/components/about.e2e.ts(8,12): error TS2304: Cannot find name 'element'.
app/about/components/about.e2e.ts(8,20): error TS2304: Cannot find name 'by'.
app/about/components/about.e2e.ts(11,3): error TS2304: Cannot find name 'it'.
app/about/components/about.e2e.ts(12,5): error TS2304: Cannot find name 'expect'.
app/about/components/about.e2e.ts(12,12): error TS2304: Cannot find name 'element'.
app/about/components/about.e2e.ts(12,20): error TS2304: Cannot find name 'by'.
app/about/components/about.e2e.ts(15,3): error TS2304: Cannot find name 'it'.
app/about/components/about.e2e.ts(16,5): error TS2304: Cannot find name 'element'.
app/about/components/about.e2e.ts(16,13): error TS2304: Cannot find name 'by'.
app/about/components/about.e2e.ts(17,5): error TS2304: Cannot find name 'element'.
app/about/components/about.e2e.ts(17,13): error TS2304: Cannot find name 'by'.
app/about/components/about.e2e.ts(18,5): error TS2304: Cannot find name 'expect'.
app/about/components/about.e2e.ts(18,12): error TS2304: Cannot find name 'element'.
app/about/components/about.e2e.ts(18,20): error TS2304: Cannot find name 'by'.
app/app/components/app.e2e.ts(1,1): error TS2304: Cannot find name 'describe'.
app/app/components/app.e2e.ts(3,3): error TS2304: Cannot find name 'beforeEach'.
app/app/components/app.e2e.ts(4,7): error TS2304: Cannot find name 'browser'.
app/app/components/app.e2e.ts(7,3): error TS2304: Cannot find name 'it'.
app/app/components/app.e2e.ts(8,7): error TS2304: Cannot find name 'expect'.
app/app/components/app.e2e.ts(8,14): error TS2304: Cannot find name 'browser'.
app/app/components/app.e2e.ts(11,3): error TS2304: Cannot find name 'it'.
app/app/components/app.e2e.ts(12,7): error TS2304: Cannot find name 'expect'.
app/app/components/app.e2e.ts(12,14): error TS2304: Cannot find name 'element'.
app/app/components/app.e2e.ts(12,22): error TS2304: Cannot find name 'by'.
app/app/components/app.e2e.ts(15,3): error TS2304: Cannot find name 'it'.
app/app/components/app.e2e.ts(16,7): error TS2304: Cannot find name 'expect'.
app/app/components/app.e2e.ts(16,14): error TS2304: Cannot find name 'element'.
app/app/components/app.e2e.ts(16,22): error TS2304: Cannot find name 'by'.
app/app/components/app.e2e.ts(19,3): error TS2304: Cannot find name 'it'.
app/app/components/app.e2e.ts(20,7): error TS2304: Cannot find name 'expect'.
app/app/components/app.e2e.ts(20,14): error TS2304: Cannot find name 'element'.
app/app/components/app.e2e.ts(20,22): error TS2304: Cannot find name 'by'.
app/app/components/app.e2e.ts(23,3): error TS2304: Cannot find name 'it'.
app/app/components/app.e2e.ts(24,7): error TS2304: Cannot find name 'expect'.
app/app/components/app.e2e.ts(24,14): error TS2304: Cannot find name 'element'.
app/app/components/app.e2e.ts(24,22): error TS2304: Cannot find name 'by'.
app/home/components/home.e2e.ts(1,1): error TS2304: Cannot find name 'describe'.
app/home/components/home.e2e.ts(3,3): error TS2304: Cannot find name 'beforeEach'.
app/home/components/home.e2e.ts(4,5): error TS2304: Cannot find name 'browser'.
app/home/components/home.e2e.ts(7,3): error TS2304: Cannot find name 'it'.
app/home/components/home.e2e.ts(8,7): error TS2304: Cannot find name 'expect'.
app/home/components/home.e2e.ts(8,14): error TS2304: Cannot find name 'element'.
app/home/components/home.e2e.ts(8,22): error TS2304: Cannot find name 'by'.
app/home/components/home.e2e.ts(12,3): error TS2304: Cannot find name 'it'.
app/home/components/home.e2e.ts(13,7): error TS2304: Cannot find name 'expect'.
app/home/components/home.e2e.ts(13,14): error TS2304: Cannot find name 'element'.
app/home/components/home.e2e.ts(13,22): error TS2304: Cannot find name 'by'.
app/home/components/home.e2e.ts(17,3): error TS2304: Cannot find name 'it'.
app/home/components/home.e2e.ts(18,7): error TS2304: Cannot find name 'expect'.
app/home/components/home.e2e.ts(18,14): error TS2304: Cannot find name 'element'.
app/home/components/home.e2e.ts(18,22): error TS2304: Cannot find name 'by'.
app/hot_loader_main.ts(5,1): error TS2304: Cannot find name 'System'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/core/change_detection/parser/locals.d.ts(3,14): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/core/change_detection/parser/locals.d.ts(4,42): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/core/di/provider.d.ts(436,103): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/core/di/provider.d.ts(436,135): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/core/zone/ng_zone.d.ts(2,37): error TS2304: Cannot find name 'Zone'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(1,25): error TS2304: Cannot find name 'MapConstructor'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(2,25): error TS2304: Cannot find name 'SetConstructor'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(4,27): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(4,39): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(7,9): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(8,30): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(11,43): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(12,27): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(14,23): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(15,25): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(94,41): error TS2304: Cannot find name 'Set'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(95,22): error TS2304: Cannot find name 'Set'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/collection.d.ts(96,25): error TS2304: Cannot find name 'Set'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/lang.d.ts(2,22): error TS2304: Cannot find name 'BrowserNodeGlobal'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/lang.d.ts(48,59): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/promise.d.ts(1,10): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/promise.d.ts(3,14): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/promise.d.ts(8,32): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/promise.d.ts(9,38): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/promise.d.ts(10,35): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/promise.d.ts(10,93): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/promise.d.ts(11,34): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/promise.d.ts(12,32): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/promise.d.ts(12,149): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/facade/promise.d.ts(13,43): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/platform/browser/browser_adapter.d.ts(75,33): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/angular2/src/platform/dom/dom_adapter.d.ts(84,42): error TS2304: Cannot find name 'Map'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/CoreOperators.d.ts(22,67): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/CoreOperators.d.ts(72,67): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/CoreOperators.d.ts(77,31): error TS2304: Cannot find name 'PromiseConstructor'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/CoreOperators.d.ts(77,54): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/Observable.d.ts(65,67): error TS2304: Cannot find name 'PromiseConstructor'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/Observable.d.ts(65,88): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/Observable.d.ts(72,84): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/Observable.d.ts(77,38): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/Observable.d.ts(100,66): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/Observable.d.ts(154,66): error TS2304: Cannot find name 'Promise'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/Observable.d.ts(159,31): error TS2304: Cannot find name 'PromiseConstructor'.
/Users/nathanbackup/Documents/github_projects/angular2-stuff/angular2-seed/node_modules/rxjs/Observable.d.ts(159,54): error TS2304: Cannot find name 'Promise'.
```

**Comment by gdi2290** [comment]
@NathanWalker the gulp build system isn't including typings/main.d.ts since you're not using the files provided in tsconfig

**Comment by NathanWalker** [comment]
I've been trying to become more knowledgable with `typings` but I'm a little unclear on what is needed. 
The `tsconfig.json` includes `main.d.ts`:
```
"files": [
        "typings/main.d.ts",
        "app/main.ts"
    ],
```
I'm missing something or don't quite understand what you are hinting at ;)

**Comment by gdi2290** [comment]
the problem is more of with the build system not including type definitions rather than typescript. see https://github.com/NathanWalker/angular2-seed/blob/typings/tools/tasks/build.js.dev.ts#L6...L13
and see https://github.com/gdi2290/answers-app/blob/typings/tasks/compile.ts#L7...L12

**Comment by NathanWalker** [comment]
You're the best @gdi2290 thank you! that gives me a good idea of what I need to do :+1:

**Comment by gdi2290** [comment]
this might help too https://gist.github.com/gdi2290/e544407980e261acc172

**Comment by NathanWalker** [comment]
Thanks, I'll try to clear this today, gotta head out for a bit.
I will say, I find systemjs configuration frustrating at best. webpack is better is this regard but when you get systemjs configured right, it's great.

**Comment by NathanWalker** [comment]
@gdi2290 One last question with regards to `tsconfig.json`... It's building out `main.js` correctly now but nothing else. I thought that since `main` pulled in `AppCmp`, it would fork out and build everything from there. Do I really need to add every single `.ts` file in the `files`?
```
"files": [
        "typings/main.d.ts",
        "app/main.ts",
        // everything else here really?
    ],
```
I tried using just `exclude` with only `node_modules` but that didn't work either.

**Comment by gdi2290** [comment]
you only need to include the entry files. So files that include other files will be seen by typescript. This might be a problem with tests since they normally don't include other files so much as everything being global

**Comment by NathanWalker** [comment]
Ok good to know that's what I thought. Unfortunately it doesn't seem to be working that way. The PR is updated with the changes but it only builds out `main.js`. strange...   

Once I solve this here, I should be able to help and solve it on universal as well (if you hadn't gotten to it yet).

**Comment by NathanWalker** [comment]
This is all fixed up now thanks to the amazing @gdi2290. Dev workflow verified, prod build verified, unit test workflows verified, and e2e all set to go.

The appveyor issue is what's on master already related to beta.1 issue: https://github.com/angular/angular/issues/6501

Please verify and merge @mgechev or @ludohenin when you can, thanks.

**Comment by ludohenin** [comment]
@NathanWalker @gdi2290 Awesome, will check that out tonight.

**Comment by NathanWalker** [comment]
@ludohenin @mgechev I need to fix `npm run docs`. I'll try to fix that soon and update this PR.

**Comment by NathanWalker** [comment]
It was a simple fix, updated now :+1:

**Comment by mgechev** [comment]
Thanks! I am on a trip to Malta, coming back on board on Saturday.

**Comment by NathanWalker** [comment]
Thanks @ludohenin, I'll leave this open for @mgechev to look at as well. I want to make sure he is okay with these changes as well.

**Comment by Bigous** [comment]
Just 2 cents... to typings work behind proxy you need to create a .typingsrc with proxy settings... Already submited a pull to look at http_proxy environment variable too...
Great work Nathan!

**Comment by mgechev** [comment]
@NathanWalker the upgrade of `tsd` to `typings` can be merged here https://github.com/mgechev/angular2-seed/pull/420.

**Pull request closed without merging**